src/Searching.js

```diff
@@ -17,29 +17,69 @@ limitations under the License.
 import EventIndexPeg from "./indexing/EventIndexPeg";
 import {MatrixClientPeg} from "./MatrixClientPeg";
 
-function serverSideSearch(term, roomId = undefined) {
-    let filter;
-    if (roomId !== undefined) {
-        // XXX: it's unintuitive that the filter for searching doesn't have
-        // the same shape as the v2 filter API :(
-        filter = {
-            rooms: [roomId],
+const SEARCH_LIMIT = 10;
+
+async function serverSideSearch(term, roomId = undefined, processResult = true) {
+    const client = MatrixClientPeg.get();
+
+    const filter = {
+        limit: SEARCH_LIMIT,
+    };
+
+    if (roomId !== undefined) filter.rooms = [roomId];
+
+    const body = {
+        search_categories: {
+            room_events: {
+                search_term: term,
+                filter: filter,
+                order_by: "recent",
+                event_context: {
+                    before_limit: 1,
+                    after_limit: 1,
+                    include_profile: true,
+                },
+            },
+        },
+    };
+
+    const response = await client.search({body: body});
+
+    if (processResult) {
+        const searchResult = {
+            _query: body,
```

Comment:
Because `backPaginateRoomEventSearch()` in the js-sdk uses it and I didn't want to introduce separate concepts for the mixed pagination and for the server-side only. Ideally all of this would be in the js-sdk and the js-sdk would have platform specific code.

Reply:
Ah, I see... 😖 Okay, maybe add a comment here about that for future readers...? It might be good to start a discussion with the team about whether it would indeed makes sense to move some bit to JS SDK, but of course that does not block this work here.